site/pages/api/create-consequence-network.api.ts

```diff
@@ -16,6 +16,8 @@ import {
 
 const createConsequenceNetwork = async (req: NextApiRequest, res: NextApiResponse): Promise<void> => {
     try {
+        const queryParam = req.headers.referer?.split("?")[1] || "";
```

Comment:
Do we not need to do any validation on the content of this query param? We're just getting whatever is after the ? in the URL, is there anything we can do to check its a valid query param?

Reply:
Also, as this is ran on lots of different API's, we should probably move this `req.headers.referer?.split("?")[1] || "";` into its own little function, especially if we add some form of validation.

Reply:
done